### PR TITLE
Clean up startup

### PR DESCRIPTION
**Pull request by jkempster34**
The `Startup` class is getting a bit unwieldy. Update it to use extension/static classes to help keep things organised.

We have also investigated having one `Startup` per environment, however, that was quite hard to follow.

### CODE REVIEW

**Review by ethax-ross — COMMENTED** [review]
I think I prefer this approach to the other one; it makes the single startup nice and simple/easier to read. I think its easier to get your head around than having the individual startups, even though that perhaps goes against convention.

File: GetIntoTeachingApi/AppStart/DatabaseUtility.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            var dbContext = scope.ServiceProvider.GetService<GetIntoTeachingDbContext>();
+
+            // Initial CRM sync.
+            if (!dbContext.PickListItems.Any())
+            {
+                RecurringJob.Trigger(JobConfiguration.CrmSyncJobId);
+            }
+
+            // Initial locations sync.s
```
Comment (ethax-ross) [inline GetIntoTeachingApi/AppStart/DatabaseUtility.cs:32]: Typo

File: GetIntoTeachingApi/AppStart/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                });
+
+                c.OperationFilter<AuthOperationFilter>();
+                c.EnableAnnotations();
+            });
+
+            services.AddFluentValidationRulesToSwagger();
+        }
+
+        public static void AddApiHangfire(this IServiceCollection services, IEnv env, bool useMemoryStorage)
```
Comment (ethax-ross) [inline GetIntoTeachingApi/AppStart/ServiceCollectionExtensions.cs:112]: Just `AddHangfire`? not sure what the `Api` references?

File: GetIntoTeachingApi/AppStart/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            services.AddSingleton<IRequestResponseLoggingConfiguration, RequestResponseLoggingConfiguration>();
+        }
+
+        public static void AddDatabase(this IServiceCollection services, IEnv env)
+        {
+            var connectionString = DbConfiguration.DatabaseConnectionString(env);
+            services.AddDbContext<GetIntoTeachingDbContext>(b => DbConfiguration.ConfigPostgres(connectionString, b));
+        }
+
+        public static void AddApiAuthentication(this IServiceCollection services)
```
Comment (ethax-ross) [inline GetIntoTeachingApi/AppStart/ServiceCollectionExtensions.cs:67]: `AddApiClientAuthentication`?

File: GetIntoTeachingApi/AppStart/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            }
+
+            app.UseAuthentication();
+
+            app.UseHttpsRedirection();
+
+            app.UseRequestResponseLogging();
+
+            app.ConfigureHangfire(
+                workerCount: !_env.IsDevelopment ? 20 : null,
```
Comment (ethax-ross) [inline GetIntoTeachingApi/AppStart/Startup.cs:97]: Reverse this to make it easier to read?

```
env.IsDevelopment ? null : 20;
```

**Review by ethax-ross — COMMENTED** [review]
Couple minor comments but this is _way_ nicer 💯

**Comment by jkempster34** [comment]
> Couple minor comments but this is _way_ nicer 💯

Thanks 🥇 All fixed.

**Review by ethax-ross — APPROVED** [review]
Looks good 👍
